src/main/java/com/example/demo/controller/BookController.java

```diff
@@ -12,16 +13,17 @@
 @RequestMapping("/book")
 public class BookController {
 
-    private Map<Long, Book> map = new HashMap<>();
+//    private Map<Long, Book> map = new HashMap<>();
+    BookService bookService=new BookService();
 
     @GetMapping("/all")
     public ResponseEntity<Collection<Book>> getAll() {
-        return ResponseEntity.ok(map.values());
+        return ResponseEntity.ok(bookService.getMap().values());
     }
 
     @GetMapping("/{id}")
     public ResponseEntity<Book> getById(@PathVariable long id) {
-        Book book = map.get(id);
+        Book book = bookService.getMap().get(id);
```

Comment:
```
bookService.getMap().get(id)
```
this should be part of Service layer, controller just need to pass id and get instance of Book